File: snippets/go/00-connection/01-login.go
// Uses shared/go/elo.go. The runner copies it and creates a temporary module.
// ELOPG_PLAN: [{"method":"login","params":{}}]
package main

import (
    "fmt"
    "example.com/elopg/elo"
)

func main() {
    client := elo.Connect() // ELOPG_* overrides the local teaching defaults.
    result, err := client.Login()
    if err != nil { panic(err) }
    fmt.Println(string(result))
}

File: snippets/java/00-connection/01-login.java
// The runner compiles this file with shared/java/EloClient.java.
// ELOPG_PLAN: [{"method":"login","params":{}}]
public final class Main {
  public static void main(String[] args) throws Exception {
    var elo = EloClient.connect(); // ELOPG_* overrides local teaching defaults.
    System.out.println(elo.call("login", "{}"));
  }
}
